### PR TITLE
Rustdoc: Fix stab disappearing and exclude cfg "doc" and "doctest"

### LINKED ISSUE #98065: Remove `cfg(doc)` and `cfg(doctest)` from `doc_auto_cfg`, fix `not()` making badge disappear
Context: https://github.com/rust-lang/rust/issues/43781#issuecomment-1154226733

CC @GuillaumeGomez

### PR DESCRIPTION
**Pull request by SpriteOvO**
Fixes #98065 Context: https://github.com/rust-lang/rust/issues/43781#issuecomment-1154226733

r? @GuillaumeGomez

### CODE REVIEW

File: src/test/rustdoc/doc-auto-cfg.rs [hunk 1]
```diff
@@ -1,14 +1,40 @@
 #![feature(doc_auto_cfg)]
-
 #![crate_name = "foo"]
 
 // @has foo/fn.foo.html
-// @has - '//*[@class="item-info"]/*[@class="stab portability"]' 'non-doctest'
-#[cfg(not(doctest))]
+// @has - '//*[@class="item-info"]/*[@class="stab portability"]' 'non-meowmeow'
+// compile-flags:--cfg feature="meowmeow"
```
Comment (GuillaumeGomez) [inline src/test/rustdoc/doc-auto-cfg.rs:6]: No need to put this one multiple times. Also, please put it at the top of the file.

**Comment by GuillaumeGomez** [comment]
Apart from the little nit in the test, looks good to me, thanks!

Want to take a look as well @notriddle ?

File: src/test/rustdoc/doc-auto-cfg.rs (hunk 1 above)
Comment (SpriteOvO) [reply src/test/rustdoc/doc-auto-cfg.rs:6]: Oh, turns out we don't need it at all, `meowmoew` is a *cfg* and not a *feature*. I'm going to delete it.

**Review by notriddle — CHANGES_REQUESTED** [review]
Mostly looks good. Just one little nitpick, then we merge!

File: src/librustdoc/clean/cfg.rs [hunk 2]
```diff
@@ -87,15 +88,21 @@ impl Cfg {
                 }),
             },
             MetaItemKind::List(ref items) => {
-                let sub_cfgs =
-                    items.iter().filter_map(|i| Cfg::parse_nested(i, exclude).transpose());
+                let sub_cfgs = items.iter().map(|i| Cfg::parse_nested(i, exclude).transpose());
+                let origin_len = sub_cfgs.len();
+                let sub_cfgs = sub_cfgs.filter_map(identity);
```
[the remark this answers is not in the document]
Comment (SpriteOvO) [reply src/librustdoc/clean/cfg.rs]: Just worried that things will change in the future, but no problem, the new tests should be able to detect the wrong length by then. Here you are :)

**Pull request merged**